docs/transpile/transpiler-plugins.ipynb

```diff
@@ -0,0 +1,301 @@
+    "\n",
+    "First, create an [HLSConfig](/api/qiskit/qiskit.transpiler.passes.HLSConfig) to\n",
+    "store the names of the plugins to use for various high-level objects. For example:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "ImportError",
+     "evalue": "cannot import name 'HLSConfig' from 'qiskit.transpiler.passes' (/home/kjs/.local/share/virtualenvs/documentation--fuetTj0/lib/python3.10/site-packages/qiskit/transpiler/passes/__init__.py)",
+     "output_type": "error",
+     "traceback": [
+      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
+      "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
+      "Cell \u001b[0;32mIn[7], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mqiskit\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mtranspiler\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mpasses\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m HLSConfig\n\u001b[1;32m      3\u001b[0m hls_config \u001b[38;5;241m=\u001b[39m HLSConfig(clifford\u001b[38;5;241m=\u001b[39m[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mlayers\u001b[39m\u001b[38;5;124m\"\u001b[39m], linear_function\u001b[38;5;241m=\u001b[39m[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpmh\u001b[39m\u001b[38;5;124m\"\u001b[39m])\n",
+      "\u001b[0;31mImportError\u001b[0m: cannot import name 'HLSConfig' from 'qiskit.transpiler.passes' (/home/kjs/.local/share/virtualenvs/documentation--fuetTj0/lib/python3.10/site-packages/qiskit/transpiler/passes/__init__.py)"
+     ]
+    }
+   ],
+   "source": [
+    "from qiskit.transpiler.passes import HLSConfig\n",
+    "\n",
+    "hls_config = HLSConfig(clifford=[\"layers\"], linear_function=[\"pmh\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This code cell creates a high-level synthesis configuration that uses the `layers` plugin\n",
+    "for synthesizing [Clifford](/api/qiskit/qiskit.quantum_info.Clifford#clifford) objects and the `pmh` plugin for synthesizing\n",
+    "[LinearFunction](/api/qiskit/qiskit.circuit.library.LinearFunction#linearfunction) objects. The names of the keyword arguments correspond to the \n",
+    "[`name`](/api/qiskit/qiskit.circuit.Operation#name) attribute of the [Operation](/api/qiskit/qiskit.circuit.Operation) class representing the type of object being synthesized.\n",
+    "For each high-level object, the list of given plugins are tried in sequence until one of them\n",
+    "succeeds (in the example above, each list only contains a single plugin). In addition to specifying\n",
+    "a plugin by its name, you can instead pass a `(name, options)` tuple, where the second element of the tuple is a dictionary containing options for the plugin.\n",
+    "\n",
```

Comment:
```suggestion
    "You can refer to the documentation for each plugin method to any available options for a given synthesis plugin.",
    "\n",
```

Reply:
Thanks. How would a user find the documentation for a plugin? For example, I mention `layers` for Clifford and `pmh` for linear function. But, I don't know of a good way to search the docs for the documentation of these plugins.

Reply:
Not currently, there is an open issue for this here: https://github.com/Qiskit/qiskit/issues/11675 we should hopefully have this fixed for the final 1.0.0 release.

Reply:
I see. Well, the plugin classes should at least have a docstring, right? Could you tell me the name of the plugin classes corresponding to the `layers` and `pmh` plugins, so that I can search for their API docs?

Reply:
And while you're at it, it would be great if you could also point me to the documentation of the `default` unitary synthesis plugin.